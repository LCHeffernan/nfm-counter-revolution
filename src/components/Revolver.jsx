import React from "react";
import PropTypes from "prop-types";

const Revolver = ({ handleClick }) => {
  return (
    <div className="revolver" data-testid="revolver">
      <button onClick={handleClick} type="button">
        increment index
      </button>
    </div>
  );
};

Revolver.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

export default Revolver;
