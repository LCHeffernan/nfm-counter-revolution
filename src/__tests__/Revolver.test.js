import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Revolver from "../components/Revolver";

const validProps = {
  handleClick: jest.fn(),
};

const setup = () => render(<Revolver {...validProps} />);

describe("Revolver", () => {
  it("matches snapshot", () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });

  beforeEach(() => setup());

  it("renders a button", () => {
    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  it("clicking button calls handleClick function", () => {
    const button = screen.getByRole("button");

    fireEvent.click(button);

    expect(validProps.handleClick).toHaveBeenCalled();
  });
});
