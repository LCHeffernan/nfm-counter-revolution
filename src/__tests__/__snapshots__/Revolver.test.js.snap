// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Revolver matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="revolver"
    data-testid="revolver"
  >
    <div
      class="disc"
      data-testid="disc"
    >
      Disc
    </div>
    <button
      type="button"
    >
      increment index
    </button>
  </div>
</DocumentFragment>
`;
